cli/src/pcluster/templates/slurm_builder.py

```diff
@@ -328,12 +332,18 @@ def _add_slurm_compute_fleet(self):
         # terminate_compute_fleet_custom_resource.add_depends_on()
 
     def _add_private_hosted_zone(self):
-        cluster_hosted_zone = route53.CfnHostedZone(
-            self.stack_scope,
-            "Route53HostedZone",
-            name=self.cluster_dns_domain.value_as_string,
-            vpcs=[route53.CfnHostedZone.VPCProperty(vpc_id=self.config.vpc_id, vpc_region=self._stack_region)],
-        )
+        if self._condition_custom_cluster_dns():
+            subdomain = self.stack_name
+            hosted_zone_id = self.config.scheduling.settings.dns.hosted_zone_id
+            domain_name = AWSApi.instance().route53.get_domain_name(hosted_zone_id)
+            cluster_hosted_zone = CustomDns(ref=hosted_zone_id, name=subdomain + domain_name)
+        else:
+            cluster_hosted_zone = route53.CfnHostedZone(
+                self.stack_scope,
+                "Route53HostedZone",
+                name=self.cluster_dns_domain.value_as_string,
```

Comment:
Here we have an inconsistency because the cluster_dns_domain is still <stackname>.pcluster. We should have a single place on which it is set and re-use it all the places.

the method should return <stackname>.pcluster if hosted_zone is not set or <stackname>.domain-name-from-route53 if it is set.

Reply:
Move `name=self.cluster_dns_domain.value_as_string` to `ClusterDNSDomain` parameter and used `self.cluster_dns_domain` here to be consistent